### PR TITLE
[repo+ci] Use SDK dotnet format in CI

### LINKED ISSUE #1274: Replace outdated tool dotnet-format by the built-in dotnet format
> BTW, it seems we are using an outdated tool dotnet-format instead of the built-in dotnet format subcommand in CI. The former seems to be completely confused by raw string literals.

_Originally posted by @Oberon00 in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1270#issuecomment-1640462239_

Similar changes done in autoinstrumentation repository https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2694 and 
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2614/files

### PR DESCRIPTION
**Pull request by CodeBlanch**
Fixes #1274

## Changes

* Updates the `dotnet-format` ci workflow to use the SDK version
* Cleans up new warnings manifested by `dotnet format`
* Removes `EnableAnalysis` property as it is now globally enabled in common props

### CODE REVIEW

File: examples/AspNet/Controllers/WeatherForecastController.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ public async Task<IEnumerable<WeatherForecast>> Get(int customerId)
     {
         if (customerId < 0)
         {
-            throw new ArgumentException();
+            throw new ArgumentOutOfRangeException(nameof(customerId));
```
Comment (utpilla) [inline examples/AspNet/Controllers/WeatherForecastController.cs:60]: nit: Add a message saying what's the expected range for customerId

File: examples/AspNet/Controllers/WeatherForecastController.cs [hunk 2]
```diff
@@ -181,7 +181,8 @@ private async Task RequestInvalidViaHttpClient()
 
             // This request is not available over SSL and will throw a handshake exception.
 
-            using var response = await request.GetAsync(this.Url.Content("~/subroute/10").Replace("http", "https")).ConfigureAwait(false);
+            using var response = await request.GetAsync(
+                new Uri(this.Url.Content("~/subroute/10").Replace("http", "https"))).ConfigureAwait(false);
```
Comment (utpilla) [inline examples/AspNet/Controllers/WeatherForecastController.cs:185]: nit: Probably good to extract some of these statements into their own variables.

File: .github/workflows/dotnet-format.yml [hunk 3]
```diff
@@ -23,8 +23,12 @@ jobs:
     - name: Setup dotnet
       uses: actions/setup-dotnet@v3
 
-    - name: Install format tool
-      run: dotnet tool install -g dotnet-format
+    - name: Install dependencies
+      run: dotnet restore
+
+    - name: Protobuf compile
```
Comment (utpilla) [inline .github/workflows/dotnet-format.yml:29]: Can we exclude generated files when running `dotnet format` instead?

Comment (CodeBlanch) [reply .github/workflows/dotnet-format.yml:29]: Good question. `dotnet format` actually excludes generated files by default. There is a switch `--include-generated` if you want to include them. Why I added this is the code consuming the generated protobuf classes blows up if they aren't there 😢 Here's an example of how it was failing before: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/actions/runs/6042774648/job/16398572729?pr=1339#step:4:989

File: examples/AspNet/Controllers/WeatherForecastController.cs (hunk 2 above)
Comment (CodeBlanch) [reply examples/AspNet/Controllers/WeatherForecastController.cs]: Updated

File: examples/AspNet/Controllers/WeatherForecastController.cs (hunk 1 above)
Comment (CodeBlanch) [reply examples/AspNet/Controllers/WeatherForecastController.cs]: Updated

**Review by twenzel — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Review by utpilla — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**